lib/screens/signup_screen.dart

```diff
@@ -0,0 +1,575 @@
+import 'package:flutter/material.dart';
+import '../widgets/background_container.dart';
+import '../widgets/custom_text_field.dart';
+import '../widgets/social_login_section.dart';
+import '../constants/app_constants.dart';
+import '../controllers/login_controller.dart';
+import '../utils/validators.dart';
+
+class SignupScreen extends StatefulWidget {
+  const SignupScreen({super.key});
+
+  @override
+  State<SignupScreen> createState() => _SignupScreenState();
+}
+
+class _SignupScreenState extends State<SignupScreen> {
+  final _formKey = GlobalKey<FormState>();
+  final _nameController = TextEditingController();
+  final _emailController = TextEditingController();
+  final _passwordController = TextEditingController();
+  final _confirmPasswordController = TextEditingController();
+  final _loginController = LoginController();
+  bool _acceptTerms = false;
+  bool _obscurePassword = true;
+  bool _obscureConfirmPassword = true;
+  bool _isLoading = false;
+
+  @override
+  void dispose() {
+    _nameController.dispose();
+    _emailController.dispose();
+    _passwordController.dispose();
+    _confirmPasswordController.dispose();
+    super.dispose();
+  }
+
+  void _handleSignup() async {
+    // Ensure the form validation doesn't cause UI issues
+    FocusScope.of(context).unfocus();
+
+    // Check for empty fields and show user-friendly messages
+    if (_nameController.text.trim().isEmpty) {
+      ScaffoldMessenger.of(context).showSnackBar(
+        const SnackBar(
+          content: Text('Please enter your full name'),
+          backgroundColor: Colors.red,
+          duration: Duration(seconds: 2),
+        ),
+      );
+      return;
+    }
+
+    if (_emailController.text.trim().isEmpty) {
+      ScaffoldMessenger.of(context).showSnackBar(
+        const SnackBar(
+          content: Text('Please enter your email address'),
+          backgroundColor: Colors.red,
+          duration: Duration(seconds: 2),
+        ),
+      );
```

Comment:
Create a single method that accepts context, message, and color (nullable) to be used in all occurences.